src/Nethermind/Nethermind.Core/MeasuredProgress.cs

```diff
@@ -16,6 +16,10 @@ public MeasuredProgress(ITimestamper? timestamper = null)
 
         public void Update(long value)
         {
+            if (UtcEndTime.HasValue)
+            {
+                UtcEndTime = null;
+            }
```

Comment:
I think `if` not needed, just setting null will suffice.

Reply:
fixed